client/src/components/MarketOrder.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+
+class MarketOrder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      power: '$1000.00',
+      view: 'buy',
+      shares: '',
+      estim: 0,
+      reviewOrder: 'null',
+      remaining: '',
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleEstimatedCost = this.handleEstimatedCost.bind(this);
+    this.renderReviewOrder = this.renderReviewOrder.bind(this);
+    this.handleReviewOrder = this.handleReviewOrder.bind(this);
+    this.backPress = this.backPress.bind(this);
+    this.handleBuy = this.handleBuy.bind(this);
+  }
+   
+  backPress(e) {
+    e.preventDefault();
+    this.setState({
+      reviewOrder: 'null'
+    })
+  }
+  
+  handleBuy(e) {
+    e.preventDefault();
+    let newPower = '$' + this.state.remaining;
```

Comment:
Hard to tell from just reading through this,  is this.state.remaining in the form you are expecting, with decimals that make sense to become a currency string?